permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,541 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
         Err("Granting at runtime is prohibited.".to_owned())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #![allow(clippy::restriction)]
+
+    use std::str::FromStr as _;
+
+    use super::*;
+
+    struct TestEnv {
+        alice_id: AccountId,
+        bob_id: AccountId,
+        carol_id: AccountId,
+        gold_asset_id: AssetId,
+        gold_asset_definition_id: AssetDefinitionId,
+        silver_asset_id: AssetId,
+        silver_asset_definition_id: AssetDefinitionId,
+        bronze_asset_id: AssetId,
+        bronze_asset_definition_id: AssetDefinitionId,
+        first_domain: (DomainId, Domain),
+        second_domain: (DomainId, Domain),
+        wsv: WorldStateView<World>,
+    }
+
+    impl TestEnv {
+        fn new() -> Self {
```

Comment:
With rust 1.62, we should be able to turn this into a `const` function.

Reply:
I'd remove the comments above blocks (but keep the grouping). Instead I'd add a doc-comment above the function, so people could hover and see both the assets, the domains and the relationships.

Reply:
Yes. const new for some types is going to be stabilised soon, so it should be possible to propagate those changes to here.